recipes-ni/ni-base-system-image-tests/files/known-permissions.csv

```diff
@@ -0,0 +1,15 @@
+Path,Mode,User,Group,Link Target
+/boot/bootmode,-rw-r--r--,admin,administrators,
+/boot/grub/!(recoverytool-ni-version),-rw-rw-r--,admin,ni,
+/boot/grub/recoverytool-ni-version,-r--r--r--,admin,administrators,
+/boot/runmode/!(bzImage),-rw-r--r--,admin,administrators,
+/boot/runmode/bzImage,lrwxrwxrwx,admin,administrators,bzImage-$(uname -r)
+/etc/fstab,-rw-r--r--,admin,administrators,
+/home/admin,drwxr-xr-x,admin,administrators,
+/home/lvuser,drwxrwsr-x,lvuser,ni,
+/home/root,lrwxrwxrwx,admin,administrators,/home/admin
+/home/webserv,drwxr-sr-x,webserv,ni,
+/lib/modules/$(uname -r)/!(build|kernel|source),-rw-r--r--,admin,administrators,
+/lib/modules/$(uname -r)/@(build|kernel),drwxr-xr-x,admin,administrators,
+/lib/modules/$(uname -r)/source,lrwxrwxrwx,admin,administrators,build
```

Comment:
I think that we should recursively check contents of these directories (possibly others too). Could we use the [globstar `**`](https://www.gnu.org/software/bash/manual/html_node/The-Shopt-Builtin.html) here?

Reply:
I've been thinking about this, and maybe the solution is we remove the first character of the mode string and just ignore if it's a file or directory altogether.

Reply:
That's likely insufficient. Directories use the execute bit for determining whether someone can list their contents, so they'll pretty much always have at least 0100 set (e.g. most system dirs are 0755, most config files are 0644).

Reply:
Yes, but we're not looking at the bits directly. Everything here is done after converting it to a human readable string and then comparing, so I think we'd get away with it.